app/src/main/java/com/naman14/timber/MusicService.java

```diff
@@ -2852,4 +2858,17 @@ public void run() {
         }
     }
 
+    private class Receiver extends BroadcastReceiver {
+        @Override
+        public void onReceive(Context content, Intent intent)
+        {
+            String action = intent.getAction();
+            if (AudioManager.ACTION_AUDIO_BECOMING_NOISY.equals(action)) {
+                if (PreferencesUtility.getInstance(content).pauseEnabledOnDetach()) {
+                    pause();
+                }
+            }
```

Comment:
Why have you created a new `BroadcastReceiver` instead of using the existing `mIntentReceiver`? It looks like you could just have added this code to [MusicService.handleCommandIntent](https://github.com/naman14/Timber/blob/63335f49ad41f813c264fcc113652a141e0a9bf8/app/src/main/java/com/naman14/timber/MusicService.java#L531).